[sachindavid96] iP

sachindavid96 opened the pull request:
### Duke

> "Forgetting things is inevitable; you need external help to prevent that from happening" - Sachin Kalinga

Duke helps you remember what you have and are yet to do. It is:

- Text-based
- Simple
- _Highly_ efficient

To use it, simply

1.  Click me to download it
2.  Double-click what you downloaded in the above step
3. Add your tasks
4. Manage the status of your tasks

It is **COMPLETELY FREE!**

Here are the features:

- [ ] Store tasks and their completion status
- [ ] Store deadlines and events with their dates and times
- [ ] Reminders (still a work in progress, coming soon!)

For those interested in Java programming, here is a snippet of the code responsible for creating deadlines:

` public Deadline(String description, String deadlineDate) throws InvalidDateFormatException {
        super(description);
        this.deadlineString = deadlineDate.strip();
        try {
            this.deadline = LocalDate.parse(this.deadlineString, formatter);
        } catch (DateTimeParseException e) {
           throw new InvalidDateFormatException();
        }
    }`

Commit:
Add Gradle support

Commit:
Level 1 added

Commit:
Added Request and Request List classes

Commit:
Changed Request class name to Task. Also, changed the name of the RequestList class to TaskList

Commit:
Level 4 added

Commit:
A-TextUiTesting - this commit was added

Commit:
Exception handling added

Commit:
Added ability to delete a task

Commit:
Added Enums

Zxun2 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,15 @@
+
+    @Override
+    public String toString() {
+        return (isDone? "[D][X] " : "[D][ ] ") + description + (". Deadline: "  + this.deadline);
```
minor nit about spacing which was mentioned in the coding standard.  this can be also be found in other similar methods.
```suggestion
    public String toString() {
        return (isDone ? "[D][X] " : "[D][ ] ") + description + (". Deadline: "  + this.deadline);
```

Zxun2 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,209 @@
+                        horizontalLine();
+                        userInput = scanner.nextLine();
+                        continue;
+                    }
```
the correct indentation can be configured in your IntelliJ ide. it should be something like this:
```suggestion
                switch (request) {
                case LIST:
                    horizontalLine();
                    list.printItems();
                    list.getTaskDetails();
                    horizontalLine();
                    userInput = scanner.nextLine();
                    continue;
```

Zxun2 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,209 @@
+                            list.getTaskDetails();
+                            horizontalLine();
+                            userInput = scanner.nextLine();
+                            continue;
```
may i check what is the intended behaviour here? if an invalid command is entered, does `continue` jump to the next iteration of the `while` loop or fallthrough the `switch-case`?

Zxun2 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,67 @@
+        numberUndone++;
+        numberOfTasks++;
+    }
+
```
missing header comments for **all** public methods. do take note of the same mistakes in other files as well

Zxun2 reviewed (commented):
code generally looks good. take note of some of my comments regarding coding standard violations!

Commit:
Functionality of saving the tasks to the hard disk has been added

Commit:
Added Date and Time API

gwynethguo commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,209 @@
+                            userInput = scanner.nextLine();
+                            continue;
+                        }
+                    }
+                    case DEADLINE: {
+                        String[] splitDeadline = userInput.split("/");
+                        String description = splitDeadline[0].substring(9);
+                        String deadline = splitDeadline[1];
+                        horizontalLine();
+                        list.addTask(new Deadline(description, deadline));
+                        list.getTaskDetails();
+                        horizontalLine();
+                        userInput = scanner.nextLine();
+                        continue;
+                    }
+                    case EVENT: {
+                        String[] splitTimes = userInput.split("/");
+                        String description = splitTimes[0].substring(6);
+                        String startDayTime = splitTimes[1];
+                        String endDayTime = splitTimes[2];
+                        horizontalLine();
+                        list.addTask(new Event(startDayTime, endDayTime, description));
+                        list.getTaskDetails();
+                        horizontalLine();
+                        userInput = scanner.nextLine();
+                        continue;
+                    }
+
+                    case DELETE: {
+                        horizontalLine();
+                        list.deleteTask(Integer.parseInt(userInputComponents[1]));
+                        horizontalLine();
+                        userInput = scanner.nextLine();
+                        continue;
+                    }
+                    default: {
+                        System.out.println("You may have accidentally entered in an invalid command. Please re-enter!");
+                        userInput = scanner.nextLine();
+                    }
+                }
```
There should be no indentation for case clauses.

gwynethguo commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,209 @@
+    public static void horizontalLine() {
+
+
+        for (int i = 0; i < 50; i++) {
+            char horizontalBar = '\u2015';
+            System.out.print(horizontalBar);
+
+        }
```
perhaps you can avoid using magic numbers?

gwynethguo commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,209 @@
+                } catch (StringIndexOutOfBoundsException e) {
+                    System.out.println("The description of todo cannot be empty!");
+                    userInput = scanner.nextLine();
+                    continue;
+                }
+            } else if (userInputComponents[0].equals("deadline")) {
+                String[] splitDeadline = userInput.split("/");
+                String description = splitDeadline[0].substring(9);
+                String deadline = splitDeadline[1];
+                horizontalLine();
+                list.addTask(new Deadline(description, deadline));
+                list.getTaskDetails();
+                horizontalLine();
+                userInput = scanner.nextLine();
+                continue;
+
+            } else if (userInputComponents[0].equals("event")) {
+                String[] splitTimes = userInput.split("/");
+                String description = splitTimes[0].substring(6);
+                String startDayTime = splitTimes[1];
+                String endDayTime = splitTimes[2];
+                horizontalLine();
+                list.addTask(new Event(startDayTime, endDayTime, description));
+                list.getTaskDetails();
+                horizontalLine();
+                userInput = scanner.nextLine();
+                continue;
+            } else if (userInputComponents[0].equals("delete")) {
+                horizontalLine();
+                list.deleteTask(Integer.parseInt(userInputComponents[1]));
+                horizontalLine();
+                userInput = scanner.nextLine();
+                continue;
+            } else {
+                System.out.println("You may have accidentally entered in an invalid command. Please re-enter!");
+                userInput = scanner.nextLine();
+            } */
+
+
+        }
```
perhaps can delete unused codes? to make the code easier to understand :)

gwynethguo reviewed (commented):
Overall your code mostly follows Java coding standard given in this module. There might be some minor mistakes for the coding standard. Perhaps you can also try to add Javadoc comments to the public methods incrementally :D Good job and keep up the good work!

Commit:
Added Parser, Storage, and UI classes

Commit:
Added the storage class

Commit:
Storage class has been slightly edited

Commit:
Changing how the different functionalities are divided into classes

Commit:
Storage class was enhanced

Commit:
Cleared certain files to prevent merge conflicts

Commit:
Merge branch 'master' into branch-Level-7

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/TaskList.java

Commit:
UI class was slightly edited

Commit:
UI class was re-added

Commit:
New request type was added

Commit:
Merge branch 'branch-Level-7'

Commit:
Made some changes to the Date and Time functionality

Commit:
Merge branch 'master' into branch-Level-8

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java

Commit:
Removed unwanted exceptions

Commit:
Added some explanatory comments

Commit:
Merge branch 'branch-Level-8'

Commit:
Fixed some bugs

Commit:
The classes were divided into two packages

Commit:
Merge branch 'add-gradle-support'

Commit:
Gradle support was added

Commit:
Added JUnit tests

Commit:
updated the details in the build.gradle file

Commit:
Added JavaDoc comments

Commit:
Fixed some coding standard violations

Commit:
Added the ability for the user to find a task by entering in a keyword

Commit:
Changed the findTask method to print a heading for the list of tasks found

Commit:
Merge branch 'branch-Level-9'

Commit:
Added some error handling

Commit:
Deleted some of the unused storage files

Commit:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/Task.java
#	src/main/java/duke/UI.java

Commit:
Added JavaFX dependencies to the gradle file

Commit:
Added JavaFX

Commit:
Added FXML files

Commit:
Merge branch 'master' into branch-Level-10

Commit:
Used FXML for the GUI

Commit:
Merge branch 'branch-Level-10'

Commit:
Enhanced the GUI with colors for the Dialog Boxes

Commit:
Add Assertions

Assertions were added to ensure that assumptions about the state of the
program are correct so that in the event that the assumptions do not hold,
they can be easily identified

Commit:
Make minor change to code-quality

Commit:
Remove unused methods

This reduces the clutter in the code and makes it more readable

sachindavid96 committed:
Merge pull request #2 from sachindavid96/branch-A-Assertions

Add assertions

sachindavid96 committed:
Merge pull request #3 from sachindavid96/branch-A-CodeQuality

Remove unused methods

Commit:
Add extension to help the user easily access the list of commands available

User has to type 'help' into the Command Line-based interface to obtain this assistance

Commit:
Improve the UI

Commit:
Fix some bugs

Commit:
Add screenshot of UI

Commit:
Improve application of Java principles
